### ISSUE TITLE
track down xpath issue in resource.keywords.controlled

### ISSUE THREAD

**Issue by jeanetteclark** [NCEAS/metadig-checks#443]
## Description

This check returns this error: ```20240119-13:01:47: [ERROR]: Selector 'thesaurusPresent' could not select single value with given Xpath: A location step was expected following the '/' or '//' token. [edu.ucsb.nceas.mdqengine.processor.XMLDialect]``` silently (the check appears to fail without error). Still working on figuring out the cause. The selector is:

``` 
<selector>
    <name>keywordGroups</name>
    <xpath>
           /resource/subjects |
           /*/identificationInfo/*/descriptiveKeywords |
           /eml/dataset/keywordSet
      </xpath>
    <subSelector>
      <name>thesaurusPresent</name>
      <xpath>(
        ./subject[subjectScheme/text()[normalize-space()]] or
        ./MD_Keywords/thesaurusName[not(@nilReason="unknown")] or
        ./keywordThesaurus
        )
      </xpath>
    </subSelector>
  </selector>
```
working theory is this line in metadig-engine, XMLDialect:348 `NodeList nodes = (NodeList)xpath.evaluate(selectorPath, contextNode, XPathConstants.NODESET);` is expecting a node set but the selector is returning a boolean. Remains to be seen why other checks where boolean is returned (seem? need to check) to work fine.

**Comment by mbjones** [comment]
@jeanetteclark Is the trailing parenthesis in the selector block unmatched? Maybe this is just a syntax issue?

**Comment by jeanetteclark** [comment]
I don't think so, I have looked at those parentheses making sure there aren't syntax errors for a long time. You can see the opening parentheses on the first line after the opening `<xpath>` tag. let me know if you see different though I would be happy at this point to find out its a simple syntax things

**Comment by mbjones** [comment]
Ah yes, I see it now. Sorry for the red herring.

**Comment by jeanetteclark** [comment]
no worries - based on the error it does seem like a syntax issue but after much time in the debugger I think I have ruled that out
